xarray/tests/test_datatree.py

```diff
@@ -821,23 +817,126 @@ def test_repr(self):
+            Group: /b/d
+                Dimensions:  (x: 2, y: 1)
+                Inherited coordinates:
+                  * x        (x) float64 16B 2.0 3.0
+                Dimensions without coordinates: y
+                Data variables:
+                    g        float64 8B 4.0
+            """
+        ).strip()
+        assert result == expected
+
+    def test_repr_two_children(self):
+        tree = DataTree.from_dict(
+            {
+                "/": Dataset(coords={"x": [1.0]}),
+                "/first_child": None,
+                "/second_child": Dataset({"foo": ("x", [0.0])}),
+            }
+        )
+
+        result = repr(tree)
+        expected = dedent(
+            """
+            <xarray.DataTree>
+            Group: /
+            │   Dimensions:  (x: 1)
+            │   Coordinates:
+            │     * x        (x) float64 8B 1.0
+            ├── Group: /first_child
+            └── Group: /second_child
+                    Dimensions:  (x: 1)
+                    Data variables:
+                        foo      (x) float64 8B 0.0
+            """
+        ).strip()
+        assert result == expected
+
+        result = repr(tree["first_child"])
+        expected = dedent(
+            """
+            <xarray.DataTree 'first_child'>
+            Group: /first_child
+                Dimensions:  (x: 1)
+                Inherited coordinates:
+                  * x        (x) float64 8B 1.0
+            """
+        ).strip()
+        assert result == expected
+
+        result = repr(tree["second_child"])
+        expected = dedent(
+            """
+            <xarray.DataTree 'second_child'>
+            Group: /second_child
+                Dimensions:  (x: 1)
+                Inherited coordinates:
+                  * x        (x) float64 8B 1.0
+                Data variables:
+                    foo      (x) float64 8B 0.0
+            """
+        ).strip()
+        assert result == expected
+
+    def test_repr_inherited_dims(self):
+        tree = DataTree.from_dict(
+            {
+                "/": Dataset({"foo": ("x", [1.0])}),
+                "/child": Dataset({"bar": ("y", [2.0])}),
+            }
+        )
+
+        result = repr(tree)
+        expected = dedent(
+            """
+            <xarray.DataTree>
+            Group: /
+            │   Dimensions:  (x: 1)
+            │   Dimensions without coordinates: x
+            │   Data variables:
+            │       foo      (x) float64 8B 1.0
+            └── Group: /child
+                    Dimensions:  (y: 1)
+                    Dimensions without coordinates: y
+                    Data variables:
+                        bar      (y) float64 8B 2.0
+            """
+        ).strip()
+        assert result == expected
+
+        result = repr(tree["child"])
+        expected = dedent(
+            """
+            <xarray.DataTree 'child'>
+            Group: /child
+                Dimensions:  (x: 1, y: 1)
```

Comment:
Isn't this one incorrect? `x` should not be inherited in this case, because it's defined only on parents' data variables, not parents' coordinate variables?

Reply:
We do still want to include it here, because it is included in the alignment checks. If you tried to insert an array with a different sized `x` dimension on `"child"` it would raise an error.

Reply:
Hmm. But it would raise an error basically because it didn't align with a variable that only exists in the parent (and is not inherited)? I feel like we're being a little unclear here on the distinction between "aligns with data in a node" and "aligns with whole tree".